Plugins/upenn.cbica.captk.brain.survival/src/internal/QmitkCaPTkSurvivalView.cpp

```diff
@@ -0,0 +1,374 @@
+                       "<a href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4767233\" style=\"color:red\">[LINK]</a>");
+  m_Controls.label_PaperInformation->setText(modelInfo);
+  m_Controls.label_PaperInformation->setTextFormat(Qt::RichText);
+  m_Controls.label_PaperInformation->setTextInteractionFlags(Qt::TextBrowserInteraction);
+  m_Controls.label_PaperInformation->setOpenExternalLinks(true);
+}
+
+void QmitkCaPTkSurvivalView::Activated()
+{
+
+}
+
+void QmitkCaPTkSurvivalView::Deactivated()
+{
+  // Not yet implemented
+}
+
+void QmitkCaPTkSurvivalView::Visible()
+{
+  // Not yet implemented
+}
+
+void QmitkCaPTkSurvivalView::Hidden()
+{
+  // Not yet implemented
+}
+
+int QmitkCaPTkSurvivalView::GetSizeFlags(bool width)
+{
+  if (!width)
+  {
+    return berry::Constants::MIN | berry::Constants::MAX | berry::Constants::FILL;
+  }
+  else
+  {
+    return 0;
+  }
+}
+
+int QmitkCaPTkSurvivalView::ComputePreferredSize(bool width,
+                                                          int /*availableParallel*/,
+                                                          int /*availablePerpendicular*/,
+                                                          int preferredResult)
+{
+  if (width == false)
+  {
+    return 100;
+  }
+  else
+  {
+    return preferredResult;
+  }
+}
+
+/************************************************************************/
+/* protected slots                                                      */
+/************************************************************************/
+void QmitkCaPTkSurvivalView::OnUsageComboBoxCurrentTextChanged(const QString& text)
+{
+  // Remember it
+  this->GetPreferences()->Put("SurvivalUsageComboBox", text);
+  this->GetPreferences()->Flush();
+
+  if (text == "Use Existing Model") {
+      m_Controls.m_cbModelSourceSelector->setVisible(true);
+      m_Controls.label_Source->setVisible(true);
+      if (m_Controls.m_cbModelSourceSelector->currentText() == "Custom") {
+          m_Controls.lineEdit_CustomModelDir->setVisible(true);
+          m_Controls.label_CustomModelDir->setVisible(true);
+          m_Controls.pushButton_CustomModelDirBrowse->setVisible(true);
+      }
+      else if (m_Controls.m_cbModelSourceSelector->currentText() == "CBICA CaPTk Model") {
+          m_Controls.groupBox_PaperInfo->setVisible(true);
+      }
+
+  }
+  else if (text == "Train New Model") {
+      m_Controls.m_cbModelSourceSelector->setVisible(false);
+      m_Controls.label_Source->setVisible(false);
+      m_Controls.lineEdit_CustomModelDir->setVisible(false);
+      m_Controls.label_CustomModelDir->setVisible(false);
+      m_Controls.pushButton_CustomModelDirBrowse->setVisible(false);
+      m_Controls.groupBox_PaperInfo->setVisible(false);
+
+  }
+}
+
+void QmitkCaPTkSurvivalView::OnModelSourceComboBoxCurrentTextChanged(const QString& text)
+{
+  // Remember it
+  this->GetPreferences()->Put("SurvivalSourceComboBox", text);
+  this->GetPreferences()->Flush();
```

Comment:
What is going on here? what do you mean by 'remember it'?

Reply:
This (vague) comment was in the template plugin that I copied -- I have replaced this comment with a more explanatory one. Will update momentarily.

Reply:
Fow now it's fine, but for future reference: If an issue is inter-module or inter-plugin is better to leave it out of PRs like this and open an issue about. That way the issue can be fixed for everything in a separate PR